### PR TITLE
Matrix Build

### PR DESCRIPTION
**Pull request by aequasi**
Takes a little bit longer (like another 30 minutes.......)

Pros:

* More split out
* Only the dependencies required for an adapter are installed
* Easier control of what tests are and arent ran
* Cleaner output of tests

Cons:

* Considerably slower

### CODE REVIEW

**Comment by aequasi** [comment]
(PR squashed)

File: .travis.yml [hunk 1]
```diff
@@ -1,49 +1,59 @@
 language: php
-sudo: required
-dist: precise
+sudo: true
```
Comment (Nyholm) [inline .travis.yml]: Are we sure we need this?

Comment (prisis) [inline .travis.yml]: If we dont use subsplitt, it should be false

**Comment by Nyholm** [comment]
:+1:

File: .travis.yml (hunk 1 above)
Comment (aequasi) [inline .travis.yml]: Mongo install requires it

Comment (aequasi) [inline .travis.yml]: If you can figure out how to use the [`apt` addon](https://docs.travis-ci.com/user/installing-dependencies/#Installing-Packages-with-the-APT-Addon), we can remove `sudo`

Comment (Nyholm) [inline .travis.yml]: Someone wrote that caches are only enabled when sudu: false

Comment (aequasi) [inline .travis.yml]: Yeah, just not possible right now with the Mongo server installs:

https://github.com/php-cache/cache/pull/24/files#diff-cd9e89841f228de7b138486546b36872R1

**Pull request merged**